src/include/miopen/solver.hpp

```diff
@@ -177,9 +157,73 @@ struct SolverMixin : SolverBase
     }
 };
 
-// Typedef for convolution solvers
+/// Typedef for convolution solvers
 using ConvSolver = SolverMixin<ConvolutionContext>;
 
+/// Base class for tunable solvers
+struct ConvTunableSolverBase : ConvSolver
```

Comment:
It looks like the only reason `ConvTunableSolverBase` exists is because `ConvOclBwdWrW2` uses it. Is this true?

Reply:
`ConvTunableSolverBase` is an interface for all tunable solvers

Reply:
But it is used only in `ConvTunableSolver` and in `ConvOclBwdWrW2`. It seems like `ConvTunableSolver` is the interface for all tunable solvers. EXCEPT `ConvOclBwdWrW2`.

Reply:
`ConvTunableSolver` is a class template for tunable solvers:
```c++
template <class PerformanceConfig>
struct ConvTunableSolver : ConvTunableSolverBase
```
Each tunable solver has its own `PerformanceConfig` which makes it impossible to use `ConvTunableSolver` as an interface.